tree/tree/test/TTreeCompression.cxx

```diff
@@ -0,0 +1,35 @@
+#include "TFile.h"
+#include "TTree.h"
+#include "TRandom.h"
+#include "gtest/gtest.h"
+
+
+class TTreeTest : public ::testing::Test {
+protected:
+    virtual void SetUp()
+    {
+        auto random = new TRandom(836);
+        auto file = new TFile("TTree.root", "RECREATE");
+        auto tree = new TTree("tree", "A test tree");
+        Double_t data = 0;
+        auto branch = tree->Branch("branch", &data);
+        for (Int_t ev = 0; ev < 1000; ev++) {
+            data = random->Gaus(100, 7);
+            tree->Fill();
+        }
+        file->Write();
+        delete random;
+        delete tree;
+        delete file;
+    }
+};
+
+TEST_F(TTreeTest, testDefaultCompression)
+{
+    auto file = new TFile("TTree.root");
+    auto compress = file->GetCompressionSettings();
+    
+    ASSERT_EQ(compress, 101);
```

Comment:
This only assert the default compression setting of the file and thus is not yet a "TTreeCompression" test.
Consider also setting and testing the setting of compression on the branch.
Consider checking that the compression is setting to the intended compression algorithm by accessing the TBasket and looking in the Buffer for the compression algorithm marker.

Reply:
@kamahori please don't forget about this! :+1: